mmdet/models/roi_heads/multi_instance_roi_head.py

```diff
@@ -0,0 +1,205 @@
+                    dim=1))
+            overlaps_normal, overlaps_normal_indices = overlaps_normal.sort(
+                descending=True, dim=1)
+            overlaps_ignore, overlaps_ignore_indices = overlaps_ignore.sort(
+                descending=True, dim=1)
+
+            # gt max and indices, ignore max and indices
+            max_overlaps_normal = overlaps_normal[:, :top_k].flatten()
+            gt_assignment_normal = overlaps_normal_indices[:, :top_k].flatten()
+            max_overlaps_ignore = overlaps_ignore[:, :top_k].flatten()
+            gt_assignment_ignore = overlaps_ignore_indices[:, :top_k].flatten()
+
+            # cons masks
+            fg_threshold = self.train_cfg['assigner']['pos_iou_thr']
+            ignore_label = self.train_cfg['assigner']['ignore_iof_thr']
+            bg_threshold_high = self.train_cfg['assigner']['neg_iou_thr']
+
+            ignore_assign_mask = (max_overlaps_normal < fg_threshold) * (
+                max_overlaps_ignore > max_overlaps_normal)
+            max_overlaps = (max_overlaps_normal * ~ignore_assign_mask) + (
+                max_overlaps_ignore * ignore_assign_mask)
+            gt_assignment = (gt_assignment_normal * ~ignore_assign_mask) + (
+                gt_assignment_ignore * ignore_assign_mask)
+
+            labels = batch_gt_instances[i].labels[gt_assignment]
+            fg_mask = (max_overlaps >= fg_threshold) * (labels != ignore_label)
+            bg_mask = (max_overlaps < bg_threshold_high) * (max_overlaps >= 0)
+            fg_mask = fg_mask.reshape(-1, top_k)
+            bg_mask = bg_mask.reshape(-1, top_k)
+
+            pos_max = self.train_cfg['sampler']['num'] * self.train_cfg[
+                'sampler']['pos_fraction']
+            fg_inds_mask = subsample_masks(fg_mask[:, 0], pos_max, True)
+            neg_max = self.train_cfg['sampler']['num'] - fg_inds_mask.sum()
+            bg_inds_mask = subsample_masks(bg_mask[:, 0], neg_max, True)
+            labels = labels + ~fg_mask.flatten(
+            )  # labels = labels * fg_mask.flatten()
+            keep_mask = fg_inds_mask + bg_inds_mask
+            # labels
+            labels = labels.reshape(-1, top_k)[keep_mask]
+            gt_assignment = gt_assignment.reshape(-1,
+                                                  top_k)[keep_mask].flatten()
+            target_boxes = gt_boxes[gt_assignment, :4]
+            rois = all_rois[keep_mask]
+            target_rois = rois.repeat(1, top_k).reshape(-1, all_rois.shape[-1])
+            bbox_targets = self.bbox_head.bbox_coder.encode(
+                target_rois, target_boxes)
+            bbox_targets = bbox_targets.reshape(-1, top_k * 4)
+            return_rois.append(rois)
+            return_labels.append(labels)
+            return_bbox_targets.append(bbox_targets)
+
+        losses = dict()
+        # bbox head loss
+        if self.with_bbox:
+            rois = bbox2roi(return_rois)
```

Comment:
it is suggested the logic here need same as other roi head, 
```
bbox_results = self.bbox_loss(x, sampling_results)
losses.update(bbox_results['loss_bbox'])
```

Reply:
This PR is still working in progress. It will be refactored later after aligning the training accuracy.